src/app/devices/kvm/kvm.component.ts

```diff
@@ -54,13 +56,14 @@ export class KvmComponent implements OnInit, AfterViewInit, OnDestroy {
       this.isLoading = true
       this.deviceId = params.id
     })
-    this.devicesService.stopwebSocket.subscribe(() => {
+    this.stopSocketSubscription = this.devicesService.stopwebSocket.subscribe(() => {
       this.stopKvm()
     })
 
-    this.devicesService.startwebSocket.subscribe(() => {
+    this.startSocketSubscription = this.devicesService.connectKvmSocket.subscribe(() => {
       this.init()
     })
+    this.timeInterval = setInterval(() => this.devicesService.getPowerState(this.deviceId).pipe().subscribe(), 15000)
```

Comment:
can leverage RXJS here, no need for .pipe() if you aren't piping anything

Reply:
Updated to use RxJS interval.